sparse/tests/test_coo.py

```diff
@@ -250,7 +250,128 @@ def test_elemwise_trinary(func, shape):
     y = ys.todense()
     z = zs.todense()
 
-    assert_eq(sparse.elemwise(func, xs, ys, zs), func(x, y, z))
+    fs = sparse.elemwise(func, xs, ys, zs)
+    assert isinstance(fs, COO)
+
+    assert_eq(fs, func(x, y, z))
+
+
+@pytest.mark.parametrize('shapes, func', [
+    ([
+         (2,),
+         (3, 2),
+         (4, 3, 2),
+     ], lambda x, y, z: (x + y) * z),
+    ([
+         (3,),
+         (2, 3),
+         (2, 2, 3),
+     ], lambda x, y, z: x * (y + z)),
+    ([
+         (2,),
+         (2, 2),
+         (2, 2, 2),
+     ], lambda x, y, z: x * y * z),
+    ([
+         (4,),
+         (4, 4),
+         (4, 4, 4),
+     ], lambda x, y, z: x + y + z),
+    ([
+         (4,),
+         (4, 4),
+         (4, 4, 4),
+     ], lambda x, y, z: x + y - z),
+    ([
+         (4,),
+         (4, 4),
+         (4, 4, 4),
+     ], lambda x, y, z: x - y + z),
+])
```

Comment:
It would be good to add checks for a few more variations on the broadcasting logic to exercise the matching logic:
- Dimensions of size 1, e.g., `(3, 1)` + `(3, 4)` -> `(3, 4)`
- Output shapes that don't match one of the inputs, e.g., `(3, 1)` + `(1, 4)` -> `(3, 4)`.
- Outputs that require matching across three inputs, e.g., `(1, 1, 2)` + `(1, 3, 1)` + `(4, 1, 1)` -> `(4, 3, 2)`.

Reply:
The first two were already covered in `test_broadcasting`. I renamed that to `test_binary_broadcasting` and moved it closer to these.

The third, I also added.